src/components/Incidents/api.js

```diff
@@ -48,56 +72,50 @@ export const collectIncidentsDataForApiQuery = (objects) => {
- * @param {string} groupedAlertsValues[].layer - The layer of the component.
+ * @param {Array} groupedAlertsValues - An array of alert objects, where each object represents
+ *                                      grouped alerts and contains fields like `group_id`,
+ *                                      `componentsList`, and `layerList`.
+ * @param {string} groupedAlertsValues[].group_id - The unique identifier for the alert group.
+ * @param {Array<string>} groupedAlertsValues[].componentsList - An array of component names.
+ * @param {Array<string>} groupedAlertsValues[].layerList - An array of layer names.
  *
- * @returns {string} The constructed query string combining all provided alerts.
+ * @returns {string} A Prometheus query string that includes conditions for each
+ *                   combination of `component` and `layer` in the `componentsList` and `layerList`.
+ *                   The queries are joined by the logical `or` operator.
  *
  * @example
  * const groupedAlertsValues = [
  *   {
- *     alertname: "KubeNodeNotReady",
- *     severity: "warning",
- *     component: "compute",
- *     layer: "compute",
- *     namespace: "openshift-monitoring"
- *   },
- *   {
- *     alertname: "AlertmanagerReceiversNotConfigured",
- *     severity: "warning",
- *     component: "monitoring",
- *     layer: "core",
- *     namespace: "openshift-monitoring"
- *   },
+ *     group_id: "639763dd-cf99-466d-be56-ae8b39f1351c",
+ *     componentsList: ["compute", "dns", "ingress"],
+ *     layerList: ["compute", "dns"]
+ *   }
  * ];
  *
  * const query = createAlertsQuery(groupedAlertsValues);
- * console.log(query);
  * // Outputs:
- * // (ALERTS{alertname="KubeNodeNotReady", namespace="openshift-monitoring", severity="warning"} + on () group_left (component, layer) (absent(meta{layer="compute", component="compute"}))) or
- * // (ALERTS{alertname="AlertmanagerReceiversNotConfigured", namespace="openshift-monitoring", severity="warning"} + on () group_left (component, layer) (absent(meta{layer="core", component="monitoring"})))
+ * // (ALERTS + on () group_left (component, group_id) (absent(meta{group_id="639763dd-cf99-466d-be56-ae8b39f1351c", component="compute", layer="compute"}))) or
+ * // (ALERTS + on () group_left (component, group_id) (absent(meta{group_id="639763dd-cf99-466d-be56-ae8b39f1351c", component="compute", layer="dns"}))) or
+ * // ...
  */
 export const createAlertsQuery = (groupedAlertsValues) => {
-  // Map through the grouped alerts to create individual alert strings for the query
   const alertsQuery = groupedAlertsValues
     .map((query) => {
-      const alertParts = ['alertname', 'namespace', 'name', 'severity', 'alertstate']
-        .filter((key) => query[key]) // Only include keys that are present in the query object
-        .map((key) => `${key}="${query[key]}"`)
-        .join(', ');
+      // Generate query for each combination of component and layer
```

Comment:
This query seems to be very expensive to execute. Can we execute queries lazily only per component/layer?